pkg/dbdump/dbdump.go

```diff
@@ -0,0 +1,238 @@
+package dbdump
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/noobaa/noobaa-operator/v5/pkg/options"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+	"github.com/sirupsen/logrus"
+
+	"github.com/spf13/cobra"
+)
+
+// Cmd returns a CLI command
+func Cmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "db-dump",
+		Short: "Collect db dump",
+		Run:   RunDump,
+		Args:  cobra.NoArgs,
+	}
+	cmd.Flags().String("dir", "", "collect db dump file into destination directory")
+	return cmd
+}
+
+// Collector configuration for diagnostics
+type Collector struct {
+	folderName 			string // Local folder to store the raw db dump
+	dbDumpFolderPath 	string // Local path of the folder containing the raw dump file
+	remoteFolderPath 	string // Remote folder (at pod) to store the raw db dump
+	remoteTarPath 		string // Remote path (at pod) of tar'd db dump
+	kubeconfig 			string
+	log        			*logrus.Entry
+}
+
+func newCollector(kubeconfig string) *Collector {
+    c := new(Collector)
+	c.folderName = fmt.Sprintf("%s_%d", "noobaa_db_dump", time.Now().Unix())
+	c.remoteTarPath = fmt.Sprintf("%s/%s%s", "/tmp", c.folderName, ".tar.gz")
+	c.log = util.Logger()
+	c.kubeconfig = kubeconfig
+    return c
+}
+
+// RunDump generates a tar file of a db dump
+func RunDump(cmd *cobra.Command, args []string) {
+	kubeconfig, _ := cmd.Flags().GetString("kubeconfig")
+	destDir, _ := cmd.Flags().GetString("dir")
+
+	CollectDBDump(kubeconfig, destDir)
+}
+
+// CollectDBDump exposes the funcioaloty to the diagnose mechanism
+func CollectDBDump(kubeconfig string, destDir string) {
+	c := newCollector(kubeconfig)
+
+	c.log.Println("Collecting db dump")
+
+	err := c.generateDBDump(destDir)
+	if(err == nil) {
+		err = c.exportDBDump(destDir)
+		if(err == nil) {
+			c.deleteRawResources()
+		}
+	}
+}
+
+// Create the dump resources
+func (c *Collector) generateDBDump(destDir string) error {
```

Comment:
Let's divide the 2 conditions into 2 functions.
one for the Postgres dump (`generatePostgresDump`) and one for the mongo (`generateMongoDBDump`)
then we can call those functions in this function.

Reply:
Sure.
Did the same for the other methods as well, as they diverged between postgrse and mongo as well.